### PR TITLE
Add test for password valid time

### PR DESCRIPTION
**Pull request by cliping**
Signed-off-by: Liping Cheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:
#avocado run --vt-type libvirt graphics.functional.positive_tests.vnc_only.set_passwd
JOB ID     : f316d42a0e6da8db37c26c6b16b1ef5a5fe9632e
JOB LOG    : /root/avocado/job-results/job-2019-03-29T19.05-f316d42/job.log
 (1/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.guest_xml: PASS (12.06 s)
 (2/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.qemu_conf: PASS (13.12 s)
 (3/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.set_passwd.valid_time: PASS (74.86 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 103.55 s

#avocado run --vt-type libvirt graphics.functional.positive_tests.spice_only.set_passwd
JOB ID     : ad2536981a8f36d81e6f518e5b7726eeb162c9dc
JOB LOG    : /root/avocado/job-results/job-2019-03-29T19.07-ad25369/job.log
 (1/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.guest_xml: PASS (14.57 s)
 (2/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.qemu_conf: PASS (12.83 s)
 (3/3) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.set_passwd.valid_time: PASS (75.85 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 106.75 s

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 1]
```diff
@@ -1018,28 +1035,34 @@ def block_ports(params):
...
     Use remote-viewer to connect guest
 
     :param graphic_type: graphic type
     :param port: port value
     :param graphic_passwd: graphic password
-    :param rv_log_str: remote-viewer debug log string
     :param test: test instance
-    :return: True or False
+    :param rv_log_str: remote-viewer debug log string
+    :param rv_log_err: remote-viewer debug log string
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: err?

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 2]
```diff
@@ -1064,23 +1087,56 @@ def check_connection(graphic_type, port, graphic_passwd, rv_log_str, test):
...
+                test.fail('After password expired, connection closed.')
 
-    if os.path.exists(virt_viewer_file):
-        os.remove(virt_viewer_file)
+            # After password expired, reconnect guest
+            rv_start = threading.Thread(target=run_remote_viewer,
+                                        args=(virt_viewer_file, opt_str))
+        else:
+            rv_start = threading.Thread(target=run_remote_viewer,
+                                        args=(virt_viewer_file, opt_str))
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: no need "else", just one line with less indent.

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 3]
```diff
@@ -1064,23 +1087,56 @@ def check_connection(graphic_type, port, graphic_passwd, rv_log_str, test):
...
+                logging.info("Keep connection after password expired.")
+
+            # stop remote-viewer command
+            utils_misc.kill_process_by_pattern("remote-viewer")
 
-    # stop remote-viewer command
-    utils_misc.kill_process_by_pattern("remote-viewer")
+            rv_result = q.get()
+            # check review-viewer log
+            if rv_log_str in rv_result and rv_log_err not in rv_result:
```
Comment (kylazhang) [inline libvirt/tests/src/graphics/graphics_functional.py]: Better with () to make it clear.

**Comment by cliping** [comment]
@kylazhang , updated, add set_passwd_valid_time().

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 4]
```diff
@@ -859,6 +860,23 @@ def check_vnc_result(vnc_opts, expected_result, all_ips, test):
     check_addr_port(all_ips, expected_vnc_ips, expected_vnc_ports, test)
 
 
+def set_passwd_valid_time(graphic, valid_time, graphic_passwd):
```
Comment (chunfuwen) [inline libvirt/tests/src/graphics/graphics_functional.py]: change name set_passwd_valid_time to set_passwd_valid_time_in_graphic

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 5]
```diff
@@ -859,6 +860,23 @@ def check_vnc_result(vnc_opts, expected_result, all_ips, test):
...
+    """
+    Set password valid time
+
+    :param graphic: a graphics device XML
+    :param valid_time: password valid time
+    :param graphic_passwd: graphic password
+    :return graphic: a graphics device XML
+    """
+    if valid_time:
```
Comment (chunfuwen) [inline libvirt/tests/src/graphics/graphics_functional.py]: do we need assert graphic_passwd !="" and none?

**Pull request merged**